spring-context/src/main/java/io/micronaut/spring/context/factory/MicronautBeanFactory.java

```diff
@@ -460,7 +460,9 @@ public Class<?> getType(@NonNull String beanName) throws NoSuchBeanDefinitionExc
                     if (beanClass != null && FactoryBean.class.isAssignableFrom(beanClass)) {
                         if (!BeanFactoryUtils.isFactoryDereference(beanName)) {
                             // If it's a FactoryBean, we want to look at what it creates, not at the factory class.
-                            return super.getTypeForFactoryBean(beanName, mbd);
+                            boolean allowInit = true; // if allowInit is true a full creation of the FactoryBean is used as fallback (through delegation to the superclass's implementation).
+                            ResolvableType resolvableType = super.getTypeForFactoryBean(beanName, mbd, allowInit);
+                            return resolvableType.getRawClass();
```

Comment:
not sure if this is correct. There is no method which takes only two parameters `beanName`,`mbd` any more

Reply:
probably fine